### ISSUE TITLE
Language test fails

### ISSUE THREAD

**Issue by racke** [sympa-community/sympa#195]
```
not ok 91 - set_lang(cs-CZ-lasstina)
#   Failed test 'set_lang(cs-CZ-lasstina)'
#   at t/Language.t line 254.
#          got: undef
#     expected: 'cs'
not ok 92 - set_lang(cs-lasstina): locale-independent case
#   Failed test 'set_lang(cs-lasstina): locale-independent case'
#   at t/Language.t line 254.
#          got: undef
#     expected: 'cs'
not ok 93 - set_lang(cs-CZ)
#   Failed test 'set_lang(cs-CZ)'
#   at t/Language.t line 254.
#          got: undef
#     expected: 'cs'
not ok 94 - set_lang(cs)
#   Failed test 'set_lang(cs)'
#   at t/Language.t line 254.
#          got: undef
#     expected: 'cs'
ok 95 - set_lang(en-CA): no catalog (en_CA) - fallback to en
ok 96 - set_lang(en-Shaw): no catalog (en@shaw) - fallback to en
ok 97 - set_lang(en): no catalog (en) - fallback to en
not ok 98 - set_lang(zh): macrolanguage zh
#   Failed test 'set_lang(zh): macrolanguage zh'
#   at t/Language.t line 254.
#          got: undef
#     expected: 'zh-TW'
not ok 99 - set_lang(zh-guoyu): macrolanguage zh
#   Failed test 'set_lang(zh-guoyu): macrolanguage zh'
#   at t/Language.t line 254.
#          got: undef
#     expected: 'zh-TW'
not ok 100 - set_lang(zh-TW)
#   Failed test 'set_lang(zh-TW)'
#   at t/Language.t line 254.
#          got: undef
#     expected: 'zh-TW'
not ok 101 - set_lang(zh-Hant): semi-macrolanguage zh-Hant
#   Failed test 'set_lang(zh-Hant): semi-macrolanguage zh-Hant'
#   at t/Language.t line 254.
#          got: undef
#     expected: 'zh-TW'
not ok 102 - set_lang(zh-Hant-TW)
#   Failed test 'set_lang(zh-Hant-TW)'
#   at t/Language.t line 254.
#          got: undef
#     expected: 'zh-TW'
not ok 103 - set_lang(zh-Hant-HK): semi-macrolanguage zh-Hant
#   Failed test 'set_lang(zh-Hant-HK): semi-macrolanguage zh-Hant'
#   at t/Language.t line 254.
#          got: undef
#     expected: 'zh-TW'
not ok 104 - set_lang(zh-Hant-guoyu): semi-macrolanguage zh-Hant
#   Failed test 'set_lang(zh-Hant-guoyu): semi-macrolanguage zh-Hant'
#   at t/Language.t line 254.
#          got: undef
#     expected: 'zh-TW'
not ok 105 - set_lang(zh-Hans-CN): macrolanguage zh
#   Failed test 'set_lang(zh-Hans-CN): macrolanguage zh'
#   at t/Language.t line 254.
#          got: undef
#     expected: 'zh-TW'
not ok 106 - native_name(cs)
#   Failed test 'native_name(cs)'
#   at t/Language.t line 261.
#          got: 'English'
#     expected: 'Česky'
not ok 107 - native_name(cs-CZ)
#   Failed test 'native_name(cs-CZ)'
#   at t/Language.t line 261.
#          got: 'English'
#     expected: 'Česky'
ok 108 - native_name(en)
ok 109 - native_name(en-CA): fallback to en
not ok 110 - native_name(zh-TW)
#   Failed test 'native_name(zh-TW)'
#   at t/Language.t line 261.
#          got: 'English'
#     expected: '繁體中文'
ok 111 - gettext(undef): undefined msgid
ok 112 - gettext(): empty msgid
ok 113 - gettext(lorem ipsum): unknown msgid
not ok 114 - gettext(_language_)
#   Failed test 'gettext(_language_)'
#   at t/Language.t line 273.
not ok 115 - gettext(Sun:Mon:Tue:Wed:Thu:Fri:Sat)
ok 116 - dgettext(web_help undef): undefined msgid
ok 117 - dgettext(web_help ): empty msgid
ok 118 - dgettext(web_help lorem ipsum): unknown msgid
not ok 119 - dgettext(web_help _language_)
not ok 120 - dgettext(web_help What is a mailing list?)
ok 121 - gettext_strftime(%a, %d %b %Y): POSIX strftime
not ok 122 - gettext_strftime(%a, %d %b %Y): emulated strftime
ok 123 - !is_utf8 gettext_strftime(%a, %d %b %Y)
#          got: 'English'
#     expected: 'Česky'
#   Failed test 'gettext(Sun:Mon:Tue:Wed:Thu:Fri:Sat)'
#   at t/Language.t line 273.
#          got: 'Sun:Mon:Tue:Wed:Thu:Fri:Sat'
#     expected: 'Ne:Po:Út:St:Čt:Pá:So'
#   Failed test 'dgettext(web_help _language_)'
#   at t/Language.t line 283.
#          got: 'English'
#     expected: 'Česky'
#   Failed test 'dgettext(web_help What is a mailing list?)'
#   at t/Language.t line 283.
#          got: 'What is a mailing list?'
#     expected: 'Co je mail list?'
#   Failed test 'gettext_strftime(%a, %d %b %Y): emulated strftime'
#   at t/Language.t line 300.
#          got: 'Thu, 01 Jan 1970'
#     expected: 'Čt 01. Led 1970'
# Looks like you failed 20 tests of 123.
Dubious, test returned 20 (wstat 5120, 0x1400)
Failed 20/123 subtests 
```

**Comment by ikedas** [comment]
Test data under `t/locale` seem to be broken.  Probably recreating them as
``` bash
$ for po in t/locale/*/LC_MESSAGES/*.po; do msgfmt -o ${po%po}.mo $po; done
```
may fix it.

**Comment by racke** [comment]
Just needs a simple correction to the command:

`for po in t/locale/*/LC_MESSAGES/*.po; do msgfmt -o ${po%po}mo $po; done
`

After running this, the test passes.

**Comment by ikedas** [comment]
Probably I am right. racke, can you check if test data includded in the source you checked out are modified?

**Comment by racke** [comment]
I am working directly from the Git repository and in fact the '*.mo' files are missing there - which is correct so far as dynamically generated files doesn't belong there.

However, neither `make` nor `make check-local` produces these files and thus the test fails.

**Comment by ikedas** [comment]
I was not so right. In fact, "make check" couldn't generate *.mo files while "make dist" could.

**Comment by racke** [comment]
Fixed through #199
